### PR TITLE
Remove embedded BC dependency and fix deprecated method use

### PR DESCRIPTION
**Pull request by IrushiL**

### CODE REVIEW

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/CRLVerifierTest.java [hunk 1]
```diff
@@ -19,12 +19,26 @@
 package org.apache.synapse.transport.certificatevalidation;
 
 import junit.framework.TestCase;
+import org.apache.synapse.commons.crypto.CryptoConstants;
 import org.apache.synapse.transport.certificatevalidation.crl.CRLCache;
 import org.apache.synapse.transport.certificatevalidation.crl.CRLVerifier;import org.bouncycastle.asn1.DERObjectIdentifier;
+import org.bouncycastle.asn1.x500.X500Name;
 import org.bouncycastle.asn1.x509.*;
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/CRLVerifierTest.java]: Shall we fix `import .*` even though it's not related to your PR?

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java [hunk 2]
```diff
@@ -19,25 +19,30 @@
 package org.apache.synapse.transport.certificatevalidation;
 
 import junit.framework.TestCase;
+import org.apache.synapse.commons.crypto.CryptoConstants;
 import org.apache.synapse.transport.certificatevalidation.ocsp.OCSPCache;
 import org.apache.synapse.transport.certificatevalidation.ocsp.OCSPVerifier;
 import org.bouncycastle.asn1.ASN1ObjectIdentifier;
 import org.bouncycastle.asn1.ocsp.OCSPObjectIdentifiers;
 import org.bouncycastle.asn1.x509.*;
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Shall we fix `import .*` even though it's not related to your PR?

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java [hunk 3]
```diff
@@ -19,25 +19,30 @@
...
 import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
 import org.bouncycastle.cert.ocsp.*;
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Shall we fix `import .*` even though it's not related to your PR?

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java [hunk 4]
```diff
@@ -19,25 +19,30 @@
...
 import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.bc.BcRSAContentSignerBuilder;
 import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
 import org.bouncycastle.operator.jcajce.JcaDigestCalculatorProviderBuilder;
-import org.bouncycastle.x509.X509V3CertificateGenerator;
 
 import java.io.IOException;
 import java.lang.reflect.Method;
 import java.math.BigInteger;
 import java.security.*;
```
Comment (indikasampath2000) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Shall we fix `import .*` even though it's not related to your PR?

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/CRLVerifierTest.java (hunk 1 above)
Comment (IrushiL) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/CRLVerifierTest.java]: Fixed in PR: https://github.com/wso2/wso2-synapse/pull/1313

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java (hunk 2 above)
Comment (IrushiL) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Fixed in PR: https://github.com/wso2/wso2-synapse/pull/1313

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java (hunk 3 above)
Comment (IrushiL) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Fixed in PR: https://github.com/wso2/wso2-synapse/pull/1313

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java (hunk 4 above)
Comment (IrushiL) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/certificatevalidation/OCSPVerifierTest.java]: Fixed in PR: https://github.com/wso2/wso2-synapse/pull/1313

**Pull request merged**